contracts/index_executor/src/steps/mod.rs

```diff
@@ -27,6 +37,12 @@ pub struct Step {
 }
 
 impl Runner for Step {
+    /// @nonce: worker's nonce
+    /// where is worker?
+    /// TODO: worker is in the context, therefore nonce should be in the context
+    /// @client: rollup client
+    /// TODO: this should be in the context as well
```

Comment:
It doesn't sound necessary, the context changed every time we execute a new step of a task(), but the rollup client was created only when the query operation triggered

Reply:
if you have a context, then you should everything in it, nonce is deeply associated with worker, why on earth did you separate them?

Reply:
I think it's possible to discard the client: RollupClient parameter, whenever you want to use rollup, create it on the fly.
also, in regard to the context, its purpose is to carry variables that otherwise would reside globally, as there should be no trouble wrapping every factor inside(including rollup client, nonce). Of course, I haven't tested the conjecture yet.